batch/batch/front_end/front_end.py

```diff
@@ -502,13 +506,15 @@
                                       job_attributes_args)
 
                 await tx.execute_update('''
-UPDATE user_resources
-SET n_ready_jobs = n_ready_jobs + %s, ready_cores_mcpu = ready_cores_mcpu + %s
-WHERE user = %s;
-
-UPDATE ready_cores SET ready_cores_mcpu = ready_cores_mcpu + %s;
+INSERT INTO batches_staging (batch_id, token, n_jobs, n_ready_jobs, ready_cores_mcpu)
+VALUES (%s, %s, %s, %s, %s)
+ON DUPLICATE KEY UPDATE
+  n_jobs = n_jobs + %s,
+  n_ready_jobs = n_ready_jobs + %s,
+  ready_cores_mcpu = ready_cores_mcpu + %s;
 ''',
-                                        (n_ready, sum_ready_cores_mcpu, user, sum_ready_cores_mcpu))
+                                        (batch_id, rand_token, n_jobs, n_ready_jobs, sum_ready_cores_mcpu,
+                                         n_jobs, n_ready_jobs, sum_ready_cores_mcpu))
```

Comment:
I don't understand this comment. I think the query is correct and I tried to split the parameters into two lines so that the first 5 (top row) are the parameters in the first set of parameters and the second set are after the key update.

Reply:
This is just about formatting.  I'm suggesting:

```
  ready_cores_mcpu = ready_cores_mcpu + %s;
''',
                                        (batch_id, rand_token,
                                         n_jobs, n_ready_jobs, sum_ready_cores_mcpu,
                                         n_jobs, n_ready_jobs, sum_ready_cores_mcpu))

```

is clearer because it makes it clear the double set of duplicated parameters match up exactly.